src/test/java/jp/ikedam/jenkins/plugins/updatesitesmanager/UpdateSitesManagerJenkinsTest.java

```diff
@@ -23,624 +23,213 @@
-            input.setChecked(true);
-        }
-        selectNewSiteForm = addNewSitePage1.getFormByName("selectNewSiteForm");
-        assertNotNull("Submit with selecting a invalid value must results in returning to the same page.", selectNewSiteForm);
-        
-        selectNewSiteForm.getInputByValue(Jenkins.getInstance().getDescriptor(TestUpdateSite.class).getClass().getName()).setChecked(true);
-        HtmlPage addNewSitePage2 = submit(selectNewSiteForm);
-        
-        // Verify form
-        HtmlForm addNewSiteForm = addNewSitePage2.getFormByName("addSiteForm");
-        assertNotNull("Form page is not shown", addNewSiteForm);
-        
-        // Sites number before posting the form.
-        int before = Jenkins.getInstance().getUpdateCenter().getSites().size();
-        
-        // Post form.
-        String id = "newsite1";
-        String url = "http://localhost/update-center.json";
-        String testValue = "Some Test Value";
-        addNewSiteForm.getInputByName("_.id").setValueAttribute(id);
-        addNewSiteForm.getInputByName("_.url").setValueAttribute(url);
-        addNewSiteForm.getInputByName("_.testValue").setValueAttribute(testValue);
-        updateSitesPage = submit(addNewSiteForm);
-        
-        // Verify new site is added.
-        assertEquals("No site is added", before + 1, Jenkins.getInstance().getUpdateCenter().getSites().size());
-        TestUpdateSite site = null;
-        
-        for(UpdateSite test: Jenkins.getInstance().getUpdateCenter().getSites())
-        {
-            if(id.equals(test.getId()))
-            {
-                site = (TestUpdateSite)test;
-                break;
-            }
-        }
-        
-        assertNotNull("Added site not found", site);
-        assertEquals("Unexpected site type", TestUpdateSite.class, site.getClass());
-        assertEquals("URL is not configured", url, site.getUrl());
-        assertEquals("Test Value is not configured", testValue, site.getTestValue());
-        
-        // added site is in list.
-        assertNotNull("Added site must be listed", updateSitesPage.getAnchorByHref(((TestUpdateSite)site).getPageUrl()));
+        assumeThat("This test must run with more than one UpdateSite Descriptors registered.",
+                target.getUpdateSiteDescriptorList(), hasSize(greaterThan(1)));
+
+        JenkinsRule.WebClient wc = jRule.createWebClient();
+
+        // configure
+        HtmlForm form = wc.goTo(UpdateSitesManager.URL).getFormByName("sitesForm");
+        HtmlPage submit = jRule.submit(form);
+
+        assertThat("should be on same url after submit",
+                submit.getWebResponse().getUrl().toString(), endsWith(UpdateSitesManager.URL + "/"));
+
+        assertThat("should see all sites", jRule.getInstance().getUpdateCenter().getSites(), hasSize(2));
+        assertThat("should see custom site",
+                jRule.getInstance().getUpdateCenter().getSites(), hasItem(isA(ManagedUpdateSite.class)));

```

Comment:
Compare the attributes of each sites. (`JenkinsRule#assertEqualDataBoundBeans` will help you).
It allows you to test your jelly files are correct.